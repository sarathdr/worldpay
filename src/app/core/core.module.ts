import { APP_INITIALIZER, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ConfigService } from './services/config.service';
import { PaymentService } from './services/payment.service';
import { CartService } from './services/cart.service';
import { HttpClientModule } from '@angular/common/http';
import { FormsModule } from '@angular/forms';


export const fetchConfig = (appConfig: ConfigService) => {
  return () => appConfig.getPaymentUrls();
};

@NgModule({
  imports: [
    CommonModule,
    FormsModule
  ],
  providers: [
    ConfigService,
    PaymentService,
    HttpClientModule,
    CartService,
    {
      provide: APP_INITIALIZER,
      useFactory: fetchConfig,
      deps: [ConfigService],
      multi: true
    }
  ]
})
export class CoreModule { }
