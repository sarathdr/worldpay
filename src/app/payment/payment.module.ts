import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AuthorizeComponent } from './authorize/authorize.component';
import { AuthorizeContainer } from './authorize/authorize.container';
import { ConfirmComponent } from './confirm/confirm.component';
import { ConfirmContainer } from './confirm/confirm.container';
import { RefundComponent } from './refund/refund.component';
import { RefundContainer } from './refund/refund.container';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { MatButtonModule, MatFormFieldModule, MatInputModule, MatSelectModule } from '@angular/material';
import { AuthorizeFormComponent } from './authorize/authorize-form.component';
import { ConfirmPaymentGuard } from './guards/confirm-payment.guard';
import { ExpiryDateModule } from '../core/directives/expiry-date.module';
import { RefundPaymentGuard } from './guards/refund-payment.guard';
import { RefundSuccessComponent } from './refund/refund-success.component';

@NgModule({
  declarations: [
    AuthorizeComponent,
    AuthorizeFormComponent,
    AuthorizeContainer,
    ConfirmComponent,
    ConfirmContainer,
    RefundComponent,
    RefundSuccessComponent,
    RefundContainer
  ],
  imports: [
    CommonModule,
    FormsModule,
    RouterModule.forChild([
      {
        path: '',
        component: AuthorizeContainer
      },
      {
        path: 'confirm',
        component: ConfirmContainer,
        canActivate:[ConfirmPaymentGuard],
        canDeactivate:[ConfirmPaymentGuard]
      },
      {
        path: 'refund',
        component: RefundContainer,
        canActivate:[RefundPaymentGuard],
        canDeactivate:[RefundPaymentGuard]
      }
    ]),
    MatFormFieldModule,
    MatInputModule,
    MatSelectModule,
    ReactiveFormsModule,
    MatButtonModule,
    ExpiryDateModule
  ],
  providers:[
    ConfirmPaymentGuard,
    RefundPaymentGuard
  ]
})
export class PaymentModule { }
