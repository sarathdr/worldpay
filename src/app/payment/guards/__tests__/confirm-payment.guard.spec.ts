import { TestBed } from '@angular/core/testing';
import { PaymentService } from '../../../core/services/payment.service';
import { Router } from '@angular/router';
import { ConfirmPaymentGuard } from '../confirm-payment.guard';


describe('ConfirmPaymentGuard', () => {

  let guard;
  let hasAuthorized: jest.Mock;
  let resetAuthorize: jest.Mock;
  let navigate: jest.Mock;

  beforeEach(() => {
    hasAuthorized = jest.fn();
    resetAuthorize = jest.fn();
    navigate = jest.fn();

    TestBed.configureTestingModule({
      imports: [],
      declarations: [],
      providers: [
        ConfirmPaymentGuard,
        {
          provide: PaymentService,
          useValue: {
            hasAuthorized,
            resetAuthorize
          }
        },
        {
          provide: Router,
          useValue: {
            navigate
          }
        }
      ]
    });

    guard = TestBed.get(ConfirmPaymentGuard);
  });

  it('should activate route if authorized', () => {
    hasAuthorized.mockReturnValue(true);
    expect(guard.canActivate()).toBeTruthy();
  });

  it('should not activate route if not authorized', () => {
    hasAuthorized.mockReturnValue(false);
    navigate.mockReturnValue(Promise.resolve());
    guard.canActivate().then(value => {
      expect(value).toBeFalsy();
      expect(navigate.mock.calls).toMatchSnapshot();
    });
  });

  it('should reset state when deactivate', () => {
    guard.canDeactivate();
    expect(resetAuthorize).toHaveBeenCalled();
  });

});
