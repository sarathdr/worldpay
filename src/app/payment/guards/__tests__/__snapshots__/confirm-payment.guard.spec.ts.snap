// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPaymentGuard should not activate route if not authorized 1`] = `
Array [
  Array [
    Array [
      "home",
    ],
  ],
]
`;
